### PR TITLE
Added increment-decrement rule

### LINKED ISSUE #3556: Port tslint-microsoft-contrib's no-increment-decrement
See https://github.com/palantir/tslint/issues/1142.

I remember reading a book (JavaScript: The Good Parts? Clean Code? The Art of Unit Testing?) by one of the big influential programming folks and encountering a passage arguing in favor of this type of rule. The justification was something like these points:
* It's easy to type `+i` or `-i` instead of `--i`
* Standardizing widths of increments/decrements makes them easier to read
* It's easy to forget the difference between `++i` and `i++`

Perhaps this rule should be named `increment-decrement` and only ban the pre- operators, to offend fewer people? The first point is the only one of the three I feel would withstand a real debate.

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
Equivalent to tslint-microsoft-contrib's no-increment-decrement. Bans ++/-- unary expressions in favor of the += equivalents. Allows an `"allow-post"` option for the i++ scenario.

Fixes #3556.

#### CHANGELOG.md entry:

[new-rule] `increment-decrement`

### CODE REVIEW

**Comment by JoshuaKGoldberg** [comment]
First iteration failure: 

> `TypeError: ts.isPostfixUnaryExpression is not a function`

`tsutils` it is.

File: test/rules/increment-decrement/default/test.ts.lint [hunk 1]
```diff
@@ -0,0 +1,20 @@
+let x = 7;
+
+++x;
+~~~       [plus]
```
Comment (ajafff) [inline test/rules/increment-decrement/default/test.ts.lint]: also test the fix output

Comment (ajafff) [inline test/rules/increment-decrement/default/test.ts.lint]: There are actually some edge cases when fixing this:

```ts
a + b++;
// currently becomes
a + b += 1; // which is invalid
// should become
a + (b += 1);
```

**Comment by ajafff** [comment]
This is yet another rule that *could* be implemented with a generic `ban-syntax` rule described in #3591 and #2497
Though that doesn't mean this PR shouldn't land.

**Comment by marcelh-gh** [comment]
@JoshuaKGoldberg Hi, so this is not yet part of 5.10? Waiting for implementation using a generic ban-syntax? Thanks & Cheers, Marcel

File: test/rules/increment-decrement/default/test.ts.lint (hunk 1 above)
Comment (giladgray) [reply test/rules/increment-decrement/default/test.ts.lint]: please fix this.

File: src/rules/incrementDecrementRule.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+                enum: [OPTION_ALLOW_POST],
+                type: "string",
+            },
+            maxLength: 1,
+            minLength: 0,
+            type: "array",
+        },
+        optionsDescription: Lint.Utils.dedent`
+            If no arguments are provided, both pre- and post- unary operators are banned.
+            If \`"${OPTION_ALLOW_POST}"\` is provided, post- unary operators will be allowed.
```
Comment (giladgray) [inline src/rules/incrementDecrementRule.ts]: should be `post-unary` both times.
